lib/truffle/truffle/patching.rb

```diff
@@ -56,6 +56,16 @@ def self.find_relative_path(absolute_path)
       end
     end
 
+    can_be_gem_path = absolute_path.each_filename.to_a & ['gems', 'lib']
+    patch_candidate = PATCHES.keys.find { |c| absolute_path.sub_ext("").to_s.end_with? c }
+    if can_be_gem_path && !!patch_candidate
```

Comment:
No need for `!!`
`!can_be_gem_path.empty?`
But otherwise Array#& creates a new Hash, so I think 2 String#include? /gems/ /lib/ would be nicer.